clients/ts/signalr/src/HttpConnection.ts

```diff
@@ -79,39 +81,35 @@ export class HttpConnection implements IConnection {
                 // No need to add a connection ID in this case
                 this.url = this.baseUrl;
                 this.transport = this.constructTransport(TransportType.WebSockets);
+                // We should just call connect directly in this case.
+                // No fallback or negotiate in this case.
+                await this.transport.connect(this.url, transferFormat, this);
             } else {
                 let headers;
                 const token = this.options.accessTokenFactory();
                 if (token) {
                     headers = {
                         ["Authorization"]: `Bearer ${token}`,
                     };
+                    this.headers = headers;
                 }
 
                 const negotiatePayload = await this.httpClient.post(this.resolveNegotiateUrl(this.baseUrl), {
```

Comment:
What is this for compared to the same call in `setNegotiationResponse`?

Reply:
Missed this